packages/server/graphql/mutations/helpers/notifications/MSTeamsNotifier.ts

```diff
@@ -0,0 +1,423 @@
+    )
+  },
+
+  async endTimeLimit(meeting, team) {
+    const {facilitatorUserId: userId} = meeting
+    const {webhookUrl} = notificationChannel
+
+    const card = new AdaptiveCards.AdaptiveCard()
+    card.version = new AdaptiveCards.Version(1.2, 0)
+
+    const meetingUrl = makeAppURL(appOrigin, `meet/${meeting.id}`)
+
+    const meetingLinkColumnSet = new AdaptiveCards.ColumnSet()
+    meetingLinkColumnSet.spacing = AdaptiveCards.Spacing.ExtraLarge
+    const meetingLinkColumn = new AdaptiveCards.Column()
+    meetingLinkColumn.width = 'stretch'
+
+    const meetingLinkHeaderTextBlock = new AdaptiveCards.TextBlock(
+      'Time’s up! Advance your meeting to the next phase'
+    )
+    meetingLinkHeaderTextBlock.wrap = true
+    meetingLinkHeaderTextBlock.weight = AdaptiveCards.TextWeight.Bolder
+    const meetingLinkTextBlock = new AdaptiveCards.TextBlock()
+    meetingLinkTextBlock.text = meetingUrl
+    meetingLinkTextBlock.color = AdaptiveCards.TextColor.Accent
+    meetingLinkTextBlock.size = AdaptiveCards.TextSize.Small
+    meetingLinkTextBlock.wrap = true
+    const joinMeetingActionSet = new AdaptiveCards.ActionSet()
+    const joinMeetingAction = new AdaptiveCards.OpenUrlAction()
+    joinMeetingAction.title = 'Advance Meeting'
+    joinMeetingAction.url = meetingUrl
+    joinMeetingAction.id = 'advanceMeeting'
+
+    joinMeetingActionSet.addAction(joinMeetingAction)
+
+    meetingLinkColumn.addItem(meetingLinkHeaderTextBlock)
+    meetingLinkColumn.addItem(meetingLinkTextBlock)
+    meetingLinkColumn.addItem(joinMeetingActionSet)
+
+    meetingLinkColumnSet.addColumn(meetingLinkColumn)
+
+    card.addItem(meetingLinkColumnSet)
+
+    const adaptivecard = JSON.stringify(card.toJSON())
+    const attachments = `{"type":"message", "attachments":[{"contentType":"application/vnd.microsoft.card.adaptive","contentUrl":null, "content": ${adaptivecard}}]}`
+
+    return notifyMSTeams('MEETING_STAGE_TIME_LIMIT_END', webhookUrl, userId, team.id, attachments)
+  },
+  async integrationUpdated() {
+    const {webhookUrl, teamId, userId} = notificationChannel
+
+    const card = new AdaptiveCards.AdaptiveCard()
+    card.version = new AdaptiveCards.Version(1.2, 0)
+
+    const messageColumnSet = new AdaptiveCards.ColumnSet()
+    messageColumnSet.spacing = AdaptiveCards.Spacing.ExtraLarge
+    const messageColumn = new AdaptiveCards.Column()
+    messageColumn.width = 'stretch'
+
+    const messageTextBlock = new AdaptiveCards.TextBlock(
+      'Integration webhook configuration updated'
+    )
+    messageTextBlock.wrap = true
+    messageTextBlock.weight = AdaptiveCards.TextWeight.Bolder
+    messageColumn.addItem(messageTextBlock)
+    messageColumnSet.addColumn(messageColumn)
+    card.addItem(messageColumnSet)
+
+    const adaptivecard = JSON.stringify(card.toJSON())
+    const attachments = `{"type":"message", "attachments":[{"contentType":"application/vnd.microsoft.card.adaptive","contentUrl":null, "content": ${adaptivecard}}]}`
```

Comment:
-1 it looks like this `attachments` string is being repeated 5 times. This is an example of something that I think should be moved to a helper function so that this file is easier for the next dev to jump into